fake.go

```diff
@@ -23,11 +23,16 @@ import (
 	"regexp"
 	"sort"
 	"strings"
+	"sync"
 )
 
 // Fake is a fake implementation of Interface
 type Fake struct {
 	nftContext
+	// lock is used to protect Table and LastTransaction.
+	// When Table and LastTransaction are accessed directly, it is caller's responsibility
+	// to ensure there is no data race.
```

Comment:
That's no good... should probably make them private members, and add accessors for them that do proper locking.

Reply:
I didn't want to break existing users, but if you think it is ok, I can make them private

Reply:
later discussed on slack; just adding accessors doesn't work because you'd have to deep-copy Table. So maybe it would make more sense to expose the locking to the caller. Make the `sync.Mutex` an embedded field like the `nftContext`, and document the fact that you need to `.RLock()` it if you are going to look directly at `Table`.

Maybe `.LastTransaction` should be changed to `.GetLastTransaction()` though, with locking. (The transaction itself would be read-only at this point, so you don't have to worry about locking after you access the pointer.)

Reply:
made lock methods external, added little comments

Reply:
Not sure I like that idea of `GetLastTransaction` because it
1. is inconsistent with Table access, which is a bit strange
2. will limit potential changes in the future if we ever need to make LastTransaction not-read-only.
wdyt?